### PR TITLE
Feature/api organisations

### PR DESCRIPTION
**Pull request by JayashreeSamantasinghar**
### What does this pull request aim to achieve? 💯
API testing for org
_Please be concise and link any related issue(s):_

- [x] I have checked that no one else is working on similar changes.
- [x ] I have read the latest [**README**](README.md) and understand the development workflow.
- [x ] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [ x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x ] There are no merge conflicts.
- [x ] There are no console warnings and errors.
- [ ] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [ ] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

**Comment by JayashreeSamantasinghar** [comment]
@galeza , @jessica-lau 
Please review the PR .Please suggest do I need to add tests to check all the keys?

File: mocha/test/organisation.test.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+let invalidId = "5fba8756dd5c260038d223a3";
+let ownerId = "5fc7aa97c0cdcc003855f304";
+
+describe("GET organisations endpoint tests", () => {
+  it("sending GET request with endpoint, status : 200 OK with org details", async () => {
+    let response = await apiHelper.sendGETRequest(APP_URL, orgApiEndPoint);
+    validator.validateStatusCode(response, 200);
+  });
+
+  it("sending GET request with endpoint with valid path parameter id, status : 200 OK with details", async () => {
```
Comment (galeza) [inline mocha/test/organisation.test.js]: I have an impression that here we test another endpoint: 
​/api​/organisations​/{organisationId}

File: mocha/test/organisation.test.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+const httpStatus = require("http-status");
+const APP_URL = process.env.MOCHA_URL;
+const apiEndPoints = require("../utils/apiEndpoints");
+const apiHelper = require("../utils/apiHelper");
+const validator = require("../utils/validators");
+
+let orgApiEndPoint = apiEndPoints.organisationsEndpoint;
+let orgId = "5fc7aa97c0cdcc003855f308";
```
Comment (galeza) [inline mocha/test/organisation.test.js]: We should not use particular ObjectIds. Tests would be very volatile and flaky.
I would check if this endpoint 
​/api​/organisations​/ returns list that is not empty.

**Review by galeza — CHANGES_REQUESTED** [review]

**Comment by galeza** [comment]
@JayashreeSamantasinghar 
I think that for the endpoint ​/api​/organisations​/ it's enough to check that body is not empty and contains list of org.
I almost finished #1862 with dbHelper. I think about integration like: ask DB how many Org documents we have in Organizations collection and then send get ​/api​/organisations​/ and just compare: num from DB == num from API get endpoint.

**Comment by JayashreeSamantasinghar** [comment]
> @JayashreeSamantasinghar
> I think that for the endpoint ​/api​/organisations​/ it's enough to check that body is not empty and contains list of org.
> I almost finished #1862 with dbHelper. I think about integration like: ask DB how many Org documents we have in Organizations collection and then send get ​/api​/organisations​/ and just compare: num from DB == num from API get endpoint.

Hi @galeza 
For the endpoint /api/organisations , I automate statuscode 200 OK, but to check body is not empty, do we need to print or to check application/JSON is content type or not that is enough?

**Review by jessica-lau — CHANGES_REQUESTED** [review]
Yea for this endpoint I would just check that list of orgs exist with the key names.

File: mocha/test/organisation.test.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      APP_URL,
+      orgApiEndPoint + "/" + invalidId,
+    );
+    validator.validateStatusCodeErrorAndMessage(
+      response,
+      404,
+      "Not Found",
+      "Not Found",
+    );
+  });
```
Comment (jessica-lau) [inline mocha/test/organisation.test.js]: Testing the invalidId probably would make more sense in the /api/organisations/{organisationId} and not in this endpoint.

Comment (JayashreeSamantasinghar) [reply mocha/test/organisation.test.js]: @jessica-lau 
In https://staging.fightpandemics.work/api/documentation/index.html#/ there are some params like limits, ownerId, skip are given for this end point ,so can I take ownerId as path parameter or not? I will remove other testcases as per your suggestions

Comment (JayashreeSamantasinghar) [reply mocha/test/organisation.test.js]: @jessica-lau , @galeza 
I did the changes suggested by you . please review this.

**Review by galeza — APPROVED** [review]
LGTM

**Review by jessica-lau — APPROVED** [review]
LGTM, thanks @JayashreeSamantasinghar

**Comment by adorapham** [comment]
@galeza @jessica-lau @JayashreeSamantasinghar @idjevm , when I try to run the end 
api/organisations with Post request, I get 401-Unauthorized instead of 404-"Route POST:/api/organisations not found. Please check to see if it is a defect for the back-end. 

Otherwise, the codes look good to me
image

**Comment by galeza** [comment]
> @galeza @jessica-lau @JayashreeSamantasinghar @idjevm , when I try to run the end
> api/organisations with Post request, I get 401-Unauthorized instead of 404-"Route POST:/api/organisations not found. Please check to see if it is a defect for the back-end.
> 
> Otherwise, the codes look good to me
> image

@adorapham 
yes it works correctly.
You can use GET api/organizations without being authorized (token is not set, cookies)
BUT with POST you need to login first and set token. That's why you saw 401-Unauthorized 
There is an endpoint POST api/organizations. Check: https://staging.fightpandemics.work/api/documentation/index.html#/


@galeza Thank you for the clarification.

**Review by adorapham — APPROVED** [review]

**Comment by JayashreeSamantasinghar** [comment]
Thank you everyone for reviewing this. 
@idjevm 
can you please merge this PR if everything seems ok to you.
Thanks.

**Review by idjevm — APPROVED** [review]

**Pull request merged**